RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronIDProducer.cc

```diff
@@ -52,15 +54,23 @@ class LowPtGsfElectronIDProducer final : public edm::global::EDProducer<> {
 ////////////////////////////////////////////////////////////////////////////////
 //
 LowPtGsfElectronIDProducer::LowPtGsfElectronIDProducer(const edm::ParameterSet& conf)
-    : electrons_(consumes<edm::View<reco::GsfElectron> >(conf.getParameter<edm::InputTag>("electrons"))),
+    : usePAT_(conf.getParameter<bool>("usePAT")),
+      electrons_(),
+      patElectrons_(),
       rho_(consumes<double>(conf.getParameter<edm::InputTag>("rho"))),
-      unbiased_(consumes<edm::ValueMap<float> >(conf.getParameter<edm::InputTag>("unbiased"))),
+      unbiased_(),
       names_(conf.getParameter<std::vector<std::string> >("ModelNames")),
       passThrough_(conf.getParameter<bool>("PassThrough")),
       minPtThreshold_(conf.getParameter<double>("MinPtThreshold")),
       maxPtThreshold_(conf.getParameter<double>("MaxPtThreshold")),
       thresholds_(conf.getParameter<std::vector<double> >("ModelThresholds")),
       version_(conf.getParameter<std::string>("Version")) {
+  if (usePAT_) {
+    patElectrons_ = consumes<pat::ElectronCollection>(conf.getParameter<edm::InputTag>("electrons"));
+  } else {
+    electrons_ = consumes<reco::GsfElectronCollection>(conf.getParameter<edm::InputTag>("electrons"));
+    unbiased_ = consumes<edm::ValueMap<float> >(conf.getParameter<edm::InputTag>("unbiased"));
+  }
```

Comment:
few lines after spotted by the tests, would be good to fix this as well

`/build/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_0_X_2021-05-28-1100/src/RecoEgamma/EgammaElectronProducers/plugins/LowPtGsfElectronIDProducer.cc:98:3: warning: direct call of function EventSetupRecord::get(ESHandle&) is deprecated and should be replaced with a call to EventSetup::getHandle(ESGetToken&). To use ESGetToken see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#In_ED_module To get data with the token see https://twiki.cern.ch/twiki/bin/view/CMSPublic/SWGuideHowToGetDataFromES#Getting_data_from_EventSetup_wit [cms-esrget]`

Reply:
Ok, but can I fix this in a separate PR to master? 
Or is the change desirable in 10_6_X as well?

Reply:
@mariadalfonso ??

Reply:
this `warning: direct call of function EventSetupRecord::get(ESHandle&)` is not to a newly added code. So, IMO, this can be migrated in a follow up PR.